internal/messaging/processor.go

```diff
@@ -141,14 +150,51 @@ func (p *processor) Request(ctx context.Context, msg *Message) (*Message, error)
 	if msg.Metadata.Recipient == "" { // TODO: add address validation
 		return nil, ErrMissingRecipient
 	}
-	msg.Metadata.Cheques = nil // TODO issue and attach cheques
+	p.logger.Infof("Distributor: received a request to propagate to CMAccount %s", msg.Metadata.Recipient)
+	cmAccountRecipient := msg.Metadata.Recipient
+	// lookup for CM Account -> bot
+	botAddress := CMAccountBotMap[common.HexToAddress(msg.Metadata.Recipient)]
+
+	if botAddress == "" {
+		// if not in mapping, fetch 1 bot from CM Account
+		botAddress, err := p.identificationHandler.getSingleBotFromCMAccountAddress(common.HexToAddress(msg.Metadata.Recipient))
+		if err != nil {
+			return nil, err
+		}
+		botAddress = "@" + strings.ToLower(botAddress) + ":" + p.identificationHandler.getMatrixHost()
+		CMAccountBotMap[common.HexToAddress(msg.Metadata.Recipient)] = botAddress
+		msg.Metadata.Recipient = botAddress
+	} else {
+		msg.Metadata.Recipient = botAddress
+	}
+
+	// Mocked data for testing TODO: @VjeraTurk remove after real Cheque impelmentations
```

Comment:
I don't think that we should leave stuff like this in PRs
@havan @Noctunus wdyt?

Reply:
Removed